src/Cli/dotnet/CommonOptions.cs

```diff
@@ -100,6 +116,57 @@ public static bool VerbosityIsDetailedOrDiagnostic(this VerbosityOptions verbosi
                 verbosity.Equals(VerbosityOptions.d) ||
                 verbosity.Equals(VerbosityOptions.detailed);
         }
+
+        internal static IEnumerable<string> ResolveArchOptionToRuntimeIdentifier(string arg, ParseResult parseResult)
+        {
+            if (parseResult.BothArchAndOsOptionsSpecified())
+            {
+                return Array.Empty<string>();
+            }
+
+            return ResolveRidShorthandOptions(null, arg);
+        }
+
+        internal static IEnumerable<string> ResolveOsOptionToRuntimeIdentifier(string arg, ParseResult parseResult)
+        {
+            if (parseResult.BothArchAndOsOptionsSpecified())
+            {
+                return ResolveRidShorthandOptions(arg, parseResult.ValueForOption<string>(CommonOptions.ArchitectureOption().Aliases.First()));
+            }
+
+            return ResolveRidShorthandOptions(arg, null);
+        }
+
+        private static IEnumerable<string> ResolveRidShorthandOptions(string os, string arch)
+        {
+            var currentRid = GetCurrentRuntimeId();
+            os = string.IsNullOrEmpty(os) ? GetOsFromRid(currentRid) : os;
+            arch = string.IsNullOrEmpty(arch) ? GetArchFromRid(currentRid) : arch;
+            var properties = new string[] { $"-property:RuntimeIdentifier={os}-{arch}" };
+            return properties;
+        }
+
+        private static string GetCurrentRuntimeId()
```

Comment:
Per the [spec](https://github.com/dotnet/designs/pull/233/files#diff-c3c573f67f5f781df30107752d1fad11567a0daed7e3008b6468fb9cdf5410e1R230):

> The shorthand syntax is intended as a CLI and not MSBuild concept. For scenarios that today require specifying RIDs in project file, the intent is that users will continue to specify RIDs. If we find that there is a need for a shorthand RID syntax in project files, then we can consider extending this syntax to MSBuild.

We could pass "private" properties to MSBuild if we had to, but if we can I think we should avoid it, as we'd have to prevent them from flowing across project references as global properties the same way we do with RuntimeIdentifier.

Reply:
Alternatively - if we have existing C# code which does this today, we should share that between MSBuild and CLI. If we don't, maybe we should consider doing so. Otherwise this will be a prime target for unintentional breaks and inconsistencies.

Reply:
For MSBuild, it's a property (`NETCoreSdkRuntimeIdentifier`) which we [generate into](https://github.com/dotnet/installer/blob/c7b0e76ac551a908c14797c3b8af809e285f0cc4/src/redist/targets/GenerateBundledVersions.targets#L280) the Microsoft.NETCoreSdk.BundledVersions.props file when we're building the SDK.

Reply:
That sounds like something we could do for CLI as well - but it would require us to build the CLI as RID specific, which is not the case today.
Alternative idea - can we instead rely on the `dotnet.dll` location? It would be easier/safer than trying to compute the path. Options:
* Simply look "next to dotnet.dll" - not as robust as the other options, but definitely much better than the code here
* Through a post-build step bake the RID into the binary. This is technically tricky and could have problems with signing - but it's an option.
* Build CLI as RID specific - the best approach I think (we plan to do that eventually anyway for size reasons)

Reply:
We already also use the `NETCoreSdkRuntimeIdentifierChain.txt` file for the [workload MSBuild SDK resolver](https://github.com/dotnet/sdk/blob/main/src/Resolvers/Microsoft.NET.Sdk.WorkloadManifestReader/WorkloadResolver.cs#L29).  So I think it's fine to rely on that file.  I think looking next to dotnet.dll is a great idea to simplify this code, in what ways do you think it wouldn't be robust?